### ISSUE TITLE
Enhance ellipsis component

### ISSUE THREAD

**Issue by chenquincy** [TuSimple/naive-ui#75]
## What problem does the feature solve?

Refer to [react-ellipsis](https://react-ellipsis.quincychen.cn/).

I think these features are necessary for an Ellipsis component:

- control whether clamp
- custom ellipsis node
- end character filter

I can make a pr with pleasure if in need.

**Comment by 07akioni** [comment]
As far as I know there would be some edge cases for js based ellipsis. For example inline element is hard to deal with. (and i think it may break vue's patching phase (when user changes slot content).

I'm not planning to do it.

However if you have knowledge about how to solve those problems. PR is welcomed.

**Comment by chenquincy** [comment]
@07akioni   I agree with you, how to judge the row count(1 or multiple of text's line-height) of a non-text inline element(higher than text) is divergent. I compute row count by text's line-height and line-clamp in `react-ellipsis` to avoid the edge cases, but it can't be suit to any case actually. Never mind js slicing here, but maybe add a prop to control whether clamp could be better?

**Comment by 07akioni** [comment]
> @07akioni I agree with you, how to judge the row count(1 or multiple of text's line-height) of a non-text inline element(higher than text) is divergent. I compute row count by text's line-height and line-clamp in `react-ellipsis` to avoid the edge cases, but it can't be suit to any case actually. Never mind js slicing here, but maybe add a prop to control whether clamp could be better?

I think it's okay. If you want to use custom ellipsis, you need to ensure only text is passed in.
